include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -624,5 +624,15 @@ ERROR(ast_format_requires_dump_ast,none,
 ERROR(unknown_dump_ast_format,none,
       "unknown format '%0' requested with '-dump-ast-format'", (StringRef))
 
+ERROR(dependency_scan_unexpected_variant, none,
+      "Unexpected variant during dependency scanning on module '%0'", (StringRef))
+NOTE(dependency_scan_unexpected_variant_context_hash_note, none,
+     "First module context hash: '%0', second module context hash: '%1'", (StringRef, StringRef))
+NOTE(dependency_scan_unexpected_variant_module_map_note, none,
+     "First module module map: '%0', second module module map: '%1'", (StringRef, StringRef))
+NOTE(dependency_scan_unexpected_variant_extra_arg_note, none,
+     "%select{First|Second}0 module command-line has extra argument: '%1'", (bool, StringRef))
```

Comment:
To match our usual style, these diagnostics should start with a lowercase letter.